app/api/dao/user.py

```diff
@@ -87,62 +89,68 @@ def delete_user(user_id):
     @email_verification_required
     def get_user(user_id):
         """ Retrieves a user's profile information using a specified ID.
-        
+
         Provides the user profile of the user whose ID matches the one specified.
-        
+
         Arguments:
             user_id: The ID of the user to be searched.
-        
+
         Returns:
             The UserModel class of the user whose ID was searched, containing the public information of their profile such as bio, location, etc.
-        
+
         """
 
         return UserModel.find_by_id(user_id)
 
     @staticmethod
     def get_user_by_email(email):
         """ Retrieves a user's profile information using a specified email.
-        
+
         Provides the user profile of the user whose email matches the one specified.
-        
+
         Arguments:
             email: The email of the user to be searched.
-        
+
         Returns:
             The UserModel class of the user whose email was searched, containing the public information of their profile such as bio, location, etc.
-        
+
         """
 
         return UserModel.find_by_email(email)
 
     @staticmethod
     def get_user_by_username(username):
         """ Retrieves a user's profile information using a specified username.
-        
+
         Provides the user profile of the user whose username matches the one specified.
-        
+
         Arguments:
             username: The ID of the user to be searched.
-        
+
         Returns:
             The UserModel class of the user whose username was searched, containing the public information of their profile such as bio, location, etc.
-        
+
         """
 
         return UserModel.find_by_username(username)
 
     @staticmethod
     def list_users(user_id, is_verified=None):
+
+        # To check whether the user is available for a mentorship relation.
+        def isUserInRelation(user):
```

Comment:
this method should return boolean otherwise name is missleading maybe `IsUserAvailableForRelation(user): return not isinstance(relation, MentorshipRelationModel) and (user['need_mentoring'] or user['available_to_mentor'])`

Reply:
That would not be possible, as then we can't assign value to the user property in the lambda function. Either we have to remove the lambda function then, or just not return this value, as in this case.

Reply:
this is why I advice you to use map not filter. map function takes iterable object transform it and return iterable object.

Reply:
if this is too hard then you can do change method name to setBlaBlaBla

Reply:
> this is why I advice you to use map not filter. map function takes iterable object transform it and return iterable object.

I have used map function only, but how then we would assign the `is_available_for_relation` property. We can't assign it inside lambda! Is there any other way?

Reply:
I have changed the name.